corehq/apps/locations/permissions.py

```diff
@@ -321,3 +321,18 @@ def _inner(request, domain, loc_id, *args, **kwargs):
         return location_restricted_response(request)
 
     return require_can_edit_or_view_locations(_inner)
+
+
+def can_access_location_data(view_fn):
+    """
+    Decorator controlling a user's access to VIEW data for a specific location.
+    """
+    @wraps(view_fn)
+    def _inner(request, domain, *args, **kwargs):
+        loc_id = request.GET.get('location_id')
+        def call_view(): return view_fn(request, domain, *args, **kwargs)
+        if loc_id is not None and not user_can_access_location_id(domain, request.couch_user, loc_id):
```

Comment:
@challabeehyv  In case there is no location_id is passed can you check is the user have national level access like this : https://github.com/dimagi/commcare-hq/blob/master/custom/icds_reports/views.py#L889

Reply:
user_can_access_location_id has this check too

Reply:
But user_can_access_location_id wont be called  when loc_id will be None no? @challabeehyv

Reply:
ohh..you mean like a default check when there is no location